pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint.go

```diff
@@ -135,34 +142,45 @@ func RedirectStep(cache *Cache, stepNum int, step *corev1.Container, kubeclient
 	step.Args = GetArgs(stepNum, step.Command, step.Args)
 	step.Command = []string{BinaryLocation}
 	step.VolumeMounts = append(step.VolumeMounts, toolsMount)
+	if stepNum == 0 {
+		step.VolumeMounts = append(step.VolumeMounts, downwardMount)
+	}
```

Comment:
can we have a comment here explaining why this is happening, why it's only for step 0, what it's for, etc.?